ntbs-service/Pages/Notifications/Edit/ClinicalDetails.cshtml.cs

```diff
@@ -39,21 +39,29 @@ public ClinicalDetailsModel(INotificationService service, NtbsContext context, I
             this.auditService = auditService;
         }
 
-        public override async Task<IActionResult> OnGetAsync(int? id)
+        public override async Task<IActionResult> OnGetAsync(int? id, bool isBeingSubmitted)
         {
             Notification = await service.GetNotificationWithNotificationSitesAsync(id);
             if (Notification == null)
             {
                 return NotFound();
             }
 
-            ClinicalDetails = Notification.ClinicalDetails;
             NotificationId = Notification.NotificationId;
+            NotificationStatus = Notification.NotificationStatus;
+            Notification.SetFullValidation(NotificationStatus, isBeingSubmitted);
+            ClinicalDetails = Notification.ClinicalDetails;
 
             if (ClinicalDetails == null) {
                 ClinicalDetails = new ClinicalDetails();
             }
 
+            ClinicalDetails.SetFullValidation(Notification.NotificationStatus, isBeingSubmitted);
```

Comment:
I'm now wondering whether owned entities were the wrong choice haha....

Would it potentially be nicer to define methods on Notification like "GetClinicalDetailsWithValidationState()", and then within Notification you can just call
`var clinicalDetails = ClinicalDetails; clinicalDetails.ShouldValidateFull = ShouldValidateFull; return clinicalDetails` etc.?

This would simplify the razor page a bit and move all the knowledge of what validation needs to happen into notification. But it might just introduce extra boilerplate.
At the very least, I don't think we need to call SetFullValidation on the owned model, I'd find it clearer if we just directly set the property to be the same as notification.ShouldValidateFull.

Reply:
I agree it's a bit much going on razor pages now. However, having separate getter for each model as you said will introduce additional boilerplate, and we will still need to SetFullValidation on Notification. But happy to move handling status into Notification model if we think it will be useful in longer run

Reply:
Or maybe it should just all go into NotificationBase? So a method on NotificationBase that looks like this:

```
protected SetValidationState(Notification notification, object ownedModel, bool isBeingSubmitted)
{
     notification.SetFullValidation(Notification.NotificationStatus, isBeingSubmitted);
     ownedModel.ShouldValidateFull = notification.ShouldValidateFull;
}
```
which can then be called from the page? Then all the validation state logic is on the NotificationBase, which makes sense more sense than putting it onto Notification as I first thought, as this is only relevant when we're actually on any of these pages... what do you think?

Reply:
Yeah, putting it just on Notification is probably not right, as we want to access it form individual pages, too.
> I'm now wondering whether owned entities were the wrong choice
Crossed my mind too, but (without spending masses of time thinking about it), it feels like doing it another way would have simply made a different part awkward, so i don't think we should kick ourselves in the shins too much ;)
Not sure if method on NotificationBase makes sense, if we're just passing `this` into ownedModel anyways - that just seems like a really convoluted way to avoid abstract method!

Reply:
Sorry I don't quite understand what you mean by the last sentence? ownedModel would just be ClinicalDetails in this case

Reply:
Ops yeah, clarified that with Farman, I misunderstood what `this` was!